tasks.schema.json

```diff
@@ -0,0 +1,335 @@
+        "type": "object",
+        "additionalProperties": false,
+        "required": [
+          "name",
+          "displayName"
+        ],
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "displayName": {
+            "type": "string"
+          },
+          "isExpanded": {
+            "type": "boolean"
+          },
+          "tags": {
+            "type": "array",
+            "items": {
+              "type": "string",
+              "enum": [
+                "Preview"
+              ]
+            }
+          }
+        }
+      }
+    },
+    "demands": {
+      "type": "array",
+      "description": "Allows you to define a list of demands that a build agent requires to run this build task.",
+      "items": {
+        "type": "string"
+      }
+    },
+    "minimumAgentVersion": {
+      "type": "string",
+      "pattern": "^\\d+\\.\\d+(\\.\\d+)?$"
+    },
+    "version": {
+      "type": "object",
+      "additionalProperties": false,
+      "description": "Always update this when you release your task, so that the agents utilise the latest code.",
+      "required": [
+        "Major",
+        "Minor",
+        "Patch"
+      ],
+      "properties": {
+        "Major": {
+          "type": "number"
+        },
+        "Minor": {
+          "type": "number"
+        },
+        "Patch": {
+          "type": "number"
+        }
+      }
+    },
+    "instanceNameFormat": {
+      "type": "string",
+      "description": "This is how the task will be displayed within the build step list - you can use variable values by using $(variablename)"
+    },
+    "inputs": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "additionalProperties": false,
+        "required": [
+          "name",
+          "label",
+          "type"
+        ],
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "The variable name to use to store the user-supplied value",
+            "pattern": "^[A-Za-z][A-Za-z0-9]*$"
+          },
+          "label": {
+            "type": "string",
+            "description": "The text displayed to the user for the input label"
+          },
+          "type": {
+            "type": "string",
+            "description": "The type that dictates the control rendered to the user.",
+            "enum": [
+              "boolean",
+              "connectedService:ServiceType",
+              "connectedService:AzureRM",
+              "connectedService:Azure",
+              "connectedService:Azure:Certificate,UsernamePassword",
+              "connectedService:Chef",
+              "connectedService:ssh",
+              "connectedService:Generic",
+              "connectedService:Jenkins",
+              "connectedService:servicefabric",
+              "filePath",
+              "multiLine",
+              "pickList",
+              "radio",
+              "string"
+            ]

```

Comment:
Because you can add new "connectedService:" within an extension, I thing this enum needs to be:

            "anyOf": [
              {
                "enum": [
                  "boolean",
                  "filePath",
                  "multiLine",
                  "pickList",
                  "radio",
                  "string"
                ]
              },
              {
                "pattern": "^connectedService\\:.*$"
              }
            ]

This still gives the intellisense for the standard inputs, but also avoids validation errors for new connectedServices. Adding known connectedServices to the enum would also be reasonable.